### ISSUE TITLE
Feature Request: Provide Session-level control over Themes or Background Colors

### ISSUE THREAD

**Issue by maharwood** [rstudio/rstudio#2350]
### System details

    RStudio Edition : 
    RStudio Version : 
    OS Version      : 
    R Version       : 

### Describe the behavior you expected

Feature Request for ability to set session-specific background colors or themes.  I am frequently working on several related or completely unrelated projects, and the ability to color-code the sessions in either the Windows R-Gui or the MacOS/Linux console makes “visual grepping” much easier and faster (“visual grepping” a term is from the description of the file-icons package for the Atom editor). Themes are great, but when I change it for one session it changes for all sessions. Please excuse any oversights on my part, and thanks!

**Comment by maharwood** [comment]
Examples of functionality in an environmental context are to set production test sessions a shade of red, demo/QA sessions a shade of yellow, and sandbox/dev sessions a shade of green.  Examples in an application context are shading financial projects shades of green (apologies for reflecting an American perspective) and risk projects shades of red.  In the Windows R-Gui or *nix console the intensity of the shading can provide finer distinctions across related projects.

**Comment by blairj09** [comment]
This would be nice to control at the RStudio project level via the Project Options modal. Another use case for this is teaching. I like to develop in RStudio with a custom dark theme, but when I'm teaching or presenting I like to use the standard textmate theme so that the theme is familiar and non-distracting.

**Comment by dvanic** [comment]
I'd like to +1 this request, and also to highlight that with R 1.4 I can definitely get this to work "manually", i.e. by opening two projects (which both open with my last used settings, for example dark mode), and then changing the theme in one of them using the RStudio/Preferences menu (to light mode): the other project still "stays" the same (i.e. dark).

**Comment by maharwood** [comment]
Thank you for the update Darya!  I had transitioned almost entirely from
RStudio to VSCode with the Window Colors extension.  Initial testing shows
RStudio 1.4.1103 does support "manual" window contrasting.

**Comment by davidski** [comment]
The workaround @dvanic is alllllmost there. Unfortunately, setting a theme in a project changes the global preference, so opening a new window takes the "new" theme. What I'd really like is a base default theme that could be overridden for a single project without changing the default theme of future projects.

**Comment by davidski** [comment]
Pinging this one to see if this could be picked up. Being able to quickly differentiate between multiple open sessions by theme would be fantastic.

**Comment by kaijagahm** [comment]
Agreed--I continue to find more and more use cases for this. The teaching use case mentioned above by @blairj09 has come up recently for me as well.

**Comment by chiasinL** [comment]
👀 this request. Have similar needs as OP and others on this thread - customized dark theme for development and need a "vanilla" theme for teaching to mostly novice learners with out-of-the-box setting.

**Comment by pete-lawson** [comment]
I'd like to +1 this request as well. The teaching case @blairj09 and others have mentioned highlights the need to make the theme a project level setting. I use RStudio for teaching and development, and I teach regularly, so I find it very frustrating having to toggle between a "vanilla" theme with scaled font for teaching, and my custom dark theme, font, and pane layout for development. Either tying those settings to the project level, or creating a theme profile switcher, where all of those settings can be saved to a profile, and switched with different profiles (for example teaching or development) would be incredible.

**Comment by mikebessuille** [comment]
This has a lot of community upvotes; moving to backlog-2

**Comment by kaijagahm** [comment]
Thank you! I'm not exaggerating that this would be the number 1 quality-of-coding-life improvement for me right now, since I usually have at least 3 projects open across various monitors at the same time and frequently need a new theme for teaching. Sounds like others feel similarly.

**Comment by jorondo1** [comment]
Has this been given any thought recently ? I keep switching back and forth and sometimes have multiple projects opened, it would be so much more convenient to be able to visually differentiate them. Also for teaching, very useful to have light themes. Hopefully this gets some traction eventually!

**Comment by gtritchie** [comment]
In upcoming RStudio 2026.06 release, I've added the ability to have a per-project editor theme setting that overrides the default global setting. This is saved as a new field in the .Rproj file, but only when set to anything other than "(Default)" (i.e. use the global theme). We can consider adding additional override capabilities (e.g. font size) in the future, but this seemed like the highest value one to do first.

<img width="432" height="395" alt="Image" src="https://github.com/user-attachments/assets/8a3f1acc-4fba-447f-818b-fe298868f525" />

**Comment by gtritchie** [comment]
Merged and will be in the 2026.06 release. I'm going to do a separate change to add a master kill-switch (i.e. to ignore project-level themes) so people who don't want arbitrary projects changing their theme can turn it off.
